src/Nox.Types/Types/User/User.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Nox.TypeOptions;
+using Nox.Enums;
+using System;
+
+namespace Nox.Types;
+
+/// <summary>
+/// The user nox type.
+/// </summary>
+public sealed class User : ValueObject<string, User>
+{
+    private UserTypeOptions _userTypeOptions = new();
+
+    /// <summary>
+    /// Gets or sets the value.
+    /// </summary>
+    public override string Value { protected set => base.Value = value.ToLowerInvariant(); }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="User"/> using the specified <see cref="UserTypeOptions"/>
+    /// </summary>
+    /// <param name="value">The string to create the <see cref="User"/> with</param>
+    /// <param name="options">The <see cref="UserTypeOptions"/> containing constraints for the value object</param>
+    /// <returns></returns>
+    /// <exception cref="ValidationException">If the user value is invalid.</exception>
+    public static User From(string value, UserTypeOptions options)
+    {
+        var newObject = new User
+        {
+            Value = value,
+            _userTypeOptions = options
+        };
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Validates a <see cref="User"/> object.
+    /// </summary>
+    /// <returns>true if the <see cref="User"/> value is valid according to the default or specified <see cref="UserTypeOptions"/>.</returns>
+    /// <exception cref="NotSupportedException">If the <see cref="UserFormatType"/> is not implemented by this method.</exception>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        switch (_userTypeOptions.UserFormatType)
+        {
+            case UserFormatType.Guid:
+                if(!Guid.TryParse(Value, out var guidResult))
+                {
+                    result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox User type {Value} as it is not a valid Guid."));
+                }
+                break;
+            case UserFormatType.Email: Email.From(Value);break;
```

Comment:
I believe the requirement was not to be referencing other Nox types. Instead, we should copy the relevant implementation.